### PR TITLE
add collect metrics to the global config

### PR DESCRIPTION
**Pull request by IsaacPD**
**Merge after**:  #5281 

**Description**
- Added field `collect-metrics` to the global config
- Check global config to see if `collect-metrics` is set and set to true and show prompt with telemetry info if it is not set
- If `collect-metrics` is set to true then collect and export user metrics
- Removes requirement of setting environment variable to enable metrics

**User facing changes**
If a user does not have `collect-metrics` set in their global config, they will see a prompt that will alert them that skaffold collects telemetry when they use skaffold and link them to where they can find out more, as well as detailing how they can disable it.

From that point on if `collect-metrics` is set to true metrics will be collected and exported.

### CODE REVIEW

**Comment by tejal29** [comment]
Skaffold will opt-in metrics collection by default right?
If `collect-metrics` is not defined, then skaffold will still collect metrics?

My thinking was we should show the metrics prompt when first time user runs  any skaffold command on v1.19.0.
I would imagine this as, 
1. If global config `collect-metrics` is not present, 
- we show the prompt and then set it to true.

The next skaffold invocation, the prompt is not shown.

If users disable metrics collection by setting `collect-metrics=false`, then don't show the prompt once in two weeks or once a quarter.
Basically we don't want to prompt the user a lot.
We also need another config "DisableMetricsPrompt" to ensure users have a way to turn off any notification or future prompts to ask them to enable metrics.

**Comment by IsaacPD** [comment]
> Skaffold will opt-in metrics collection by default right?
> If `collect-metrics` is not defined, then skaffold will still collect metrics?
> 
> My thinking was we should show the metrics prompt when first time user runs any skaffold command on v1.19.0.
> I would imagine this as,
> 
> 1. If global config `collect-metrics` is not present,   we show the prompt and then set it to true.
> 2. For any future skaffold invocations,  if `collect-metrics=true`  then no prompt is shown.
> 3. For any future skaffold invocations, if  `collect-metrics=false`, then show the prompt once in two weeks or once a quarter. Basically we don't want to prompt the user a lot similar to survey prompt.
> 4. We also need another config "DisableMetricsPrompt" to ensure users have a way to turn off this 2 week notification pop up. If its set to true, dont show the prompt to enable metrics ever.

What you have outlined is how this PR would work. 1. and 2. are already satisfied, and if `collect-metrics=false` then it acts the same as 2. and no prompt is shown.

I don't think it makes sense to repeatedly prompt the user that skaffold collects metrics if they have explicitly disabled them. I'm not sure if they'd change their mind, and we would be basically asking them to opt-in after they had already opted-out.

**Comment by tejal29** [comment]
Testing notes

1. run first time - config `collect-metrics` not set -  show the prompt
```
../../out/skaffold build -d gcr.io/tejal-test
Generating tags...
 - leeroy-web -> gcr.io/tejal-test/leeroy-web:v1.18.0-11-g4d40aa090-dirty
 - leeroy-app -> gcr.io/tejal-test/leeroy-app:v1.18.0-11-g4d40aa090
 - base -> gcr.io/tejal-test/base:v1.18.0-11-g4d40aa090
Checking cache...
 - leeroy-web: Found Remotely
 - leeroy-app: Found Remotely
 - base: Found Remotely
To help improve the quality of this product, we collect anonymized usage data for details on what is tracked and how we use this data visit <https://skaffold.dev/docs/telemetry/>. This data is handled in accordance with our privacy policy <https://policies.google.com/privacy/>

You may choose to opt out of this collection by running the following command:
	skaffold config set --global collect-metrics false
```
2. Run again - no prompt shown
``` 

```
3. Set collect-metrics, false, and re-run - no prompt shown
```
../../out/skaffold build -d gcr.io/tejal-test
Generating tags...
 - leeroy-web -> gcr.io/tejal-test/leeroy-web:v1.18.0-11-g4d40aa090-dirty
 - leeroy-app -> gcr.io/tejal-test/leeroy-app:v1.18.0-11-g4d40aa090
 - base -> gcr.io/tejal-test/base:v1.18.0-11-g4d40aa090
Checking cache...
 - leeroy-web: Found Remotely
 - leeroy-app: Found Remotely
 - base: Found Remotely

```

**Review by tejal29 — APPROVED** [review]
